src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -666,6 +666,28 @@ protected int getCacheLimit() {
 
     }
 
+    @CoreMethod(names = "keys?", isModuleFunction = true, required = 1)
+    public abstract static class InteropHasKeysNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public boolean hasKeys(
+                TruffleObject receiver,
+                @Cached("createHasKeysNode()") Node hasKeysNode) {
+            return ForeignAccess.sendHasKeys(hasKeysNode, receiver);
+        }
+
+        protected Node createHasKeysNode() {
+            return Message.HAS_KEYS.createNode();
+        }
+
+        @Specialization(guards = "!isTruffleObject(receiver)")
+        public boolean hasKeys(Object receiver,
+                               @Cached("create()") HasKeysNode hasKeysNode) {
+            return hasKeysNode.executeHasKeys(receiver);
```

Comment:
Maybe simply return `false` here?
It seems weird to use the Ruby node here.

Reply:
Then I'd be implementing the logic for what classes have keys for interop in two different places.